src/transformStateWithClones.js

```diff
@@ -6,8 +6,36 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    const { type, keysToRemove, extraData } = action;
```

Comment:
You can use shorter variant

```suggestion
  for (const { type, keysToRemove, extraData } of actions) {
```